client/src/components/Workflow/Editor/Index.vue

```diff
@@ -532,30 +536,27 @@ export default {
         onDownload() {
             window.location = `${getAppRoot()}api/workflows/${this.id}/download?format=json-download`;
         },
-        async doSaveAs(create = false) {
-            const rename_name = create ? this.name : this.saveAsName ?? `SavedAs_${this.name}`;
-            const rename_annotation = create ? this.annotation || "" : this.saveAsAnnotation ?? "";
-
-            // This is an old web controller endpoint that wants form data posted...
-            const formData = new FormData();
-            formData.append("workflow_name", rename_name);
-            formData.append("workflow_annotation", rename_annotation);
-            formData.append("from_tool_form", true);
-            formData.append("workflow_data", JSON.stringify(toSimple(this.id, this)));
+        async doSaveAs() {
+            if (!this.saveAsName && !this.nameValidate()) {
+                return;
+            }
+            const rename_name = this.saveAsName ?? `SavedAs_${this.name}`;
```

Comment:
The `SavedAs` case can't actually happen because of the check on line 540, or am I misreading this ?

Reply:
You are right, i messed this up refactoring the Save as function, thanks for pointing this out

Reply:
Actually, no, that would be the case if line 540 had an OR

**line 540:**
- if user doesn't provide a _Save As name_ in the save as modal, 
- AND the user doesn't have a `this.name` (set in the Editor right panel name input field)

_return_

**Now in line 543:**
- it is still possible that the user didn't provide a _Save As name_, but they might have still had a `this.name`, so save the workflow as "SaveAs_OGWf name"